recipes/c-blosc2/all/conanfile.py

```diff
@@ -65,13 +65,13 @@ def layout(self):
 
     def requirements(self):
         if self.options.with_lz4:
-            self.requires("lz4/1.9.4")
+            self.requires("lz4/1.9.4", transitive_libs=True)
```

Comment:
Having ``transitive_libs=True``, specially without ``transitive_header=true`` should be quite exceptional, so it would be good to have here some comment (in code), explaining why it is necessary (also a bit surprising that it was not necessary before, was this recipe in its previous 2.8.0 version ready for 2.0?)

Reply:
I see that the CI is failing, linking errors when ``shared=True``.
Can you please try changing in ``test_package/CMakeLists.txt``?:
```diff
--target_link_libraries(${PROJECT_NAME} PRIVATE c-blosc2::c-blosc2)
++target_link_libraries(${PROJECT_NAME} PUBLIC c-blosc2::c-blosc2)
```
It seems that sometimes it works, because there is some missing information in the generated .cmake files due to lack of information about libs beings "shared"

Reply:
@memsharded 
Thanks! 
I try it.

Reply:
@memsharded 
Sadly, `PUBLIC` doesn't solve this issue.

About `transitive_libs=True`, it fix link errors(undefined reference) in test_package.
```
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `LZ4_compress_HC'
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `ZSTD_decompressDCtx'
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `ZSTD_isError'
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `ZSTD_freeCDict'
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `ZSTD_freeCCtx'
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `ZSTD_freeDCtx'
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `ZDICT_getErrorName'
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `ZSTD_createCCtx'
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `ZSTD_decompress_usingDDict'
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `ZSTD_compress_usingCDict'
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `ZSTD_createDDict'
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `ZSTD_compressCCtx'
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `ZSTD_createDCtx'
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `ZSTD_createCDict'
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `LZ4_decompress_safe'
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `LZ4_compress_fast'
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `ZDICT_trainFromBuffer'
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `ZSTD_freeDDict'
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `ZDICT_isError'
/usr/bin/ld: /home/toge/.conan2/p/b/c-blo8fd052ea06fd1/p/lib/libblosc2.so: undefined reference to `ZSTD_maxCLevel'
```